cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -157,6 +163,162 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             ;
     }
 
+    template <typename D /* for destination type */>
+    struct TypeBroadCast
+    {
+        static void broadcast(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            using shape_type = small_vector<size_t>;
+            using out_type = typename std::remove_reference<decltype(arr_out[0])>::type;
+
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            auto * arr_new = reinterpret_cast<pybind11::array_t<D> *>(&arr_in);
```

Comment:
This looks hacky.  I cannot think of a better way to do it, though.

Reply:
I think it's a smart way :P

Reply:
I can't say it's not :)